CONTRIBUTING.md

```diff
@@ -1,62 +1,99 @@
 # Contributing Guidelines
 
-Thank you for your interest in contributing to our project. Whether it's a bug report, new feature, correction, or additional
-documentation, we greatly value feedback and contributions from our community.
+Thank you for your interest in contributing to our project. We greatly value
+feedback and contributions from our community!
 
-Please read through this document before submitting any issues or pull requests to ensure we have all the necessary
-information to effectively respond to your bug report or contribution.
+Reviewing this document will maximize your success in working with the
+codebase and sending pull requests.
 
 ## Getting Started
 
+### Find things to do
+
+If you're looking for ideas about where to contribute, consider
+[_good first issue_](https://github.com/aws/aws-toolkit-vscode/labels/good%20first%20issue)
+issues.
+
 ### Setup
 
-Before you start, install the following dependencies.
+To develop this project, install these dependencies:
 
 -   [Visual Studio Code](https://code.visualstudio.com/Download)
 -   [NodeJS and NPM](https://nodejs.org/)
-    -   NodeJS Version: 10.x
-    -   NPM version: 6.4.1 or higher
+    -   NodeJS Version: 12.x
+    -   NPM version: 7.x or higher
 -   [Typescript](https://www.typescriptlang.org/)
 -   [Git](https://git-scm.com/downloads)
 -   [AWS `git secrets`](https://github.com/awslabs/git-secrets)
 
-Then clone the repository and install npm dependencies:
+Then clone the repository and install NPM packages:
 
     cd ~/repos
     git clone git@github.com:aws/aws-toolkit-vscode.git
     cd aws-toolkit-vscode
     npm install
 
+### Run
+
+You can run the extension from Visual Studio Code:
+
+1. Select the Run panel from the sidebar.
+2. From the dropdown at the top of the Run pane, choose `Extension`.
+3. Press `F5` to launch a new instance of Visual Studio Code with the extension installed and the debugger attached.
+
 ### Build
 
 When you launch the extension or run tests from Visual Studio Code, it will automatically build the extension and watch for changes.
 
-Alternatively, these NPM tasks are defined:
+You can also use these NPM tasks (see `npm run` for the full list):
 
--   To build one time:
+-   To build once:
     ```
     npm run compile
     ```
 -   To build and watch for file changes:
     ```
     npm run watch
     ```
--   To build a release VSIX artifact:
+-   To build a release artifact (VSIX):
     ```
     npm run package
     ```
--   To build a "debug" VSIX artifact (faster, does not minify the webpack result):
+    - This uses webpack which may exhaust the default Node heap size on Linux.
+      To fix this set `--max-old-space-size`:
+
+      ```
+      export NODE_OPTIONS=--max-old-space-size=8192
```

Comment:
Do we need to specify that this is a Linux command or provide instructions for Windows?

Reply:
updated since we already specify unix/powershell elsewhere in this document. However developers should understand how to set env vars in whatever shell they prefer.

Reply:
This issue only applies to linux ( which is mentioned in this section).